src/test/java/de/bonndan/nivio/input/GroupQueryResolverTest.java

```diff
@@ -0,0 +1,97 @@
+package de.bonndan.nivio.input;
+
+import de.bonndan.nivio.input.dto.GroupDescription;
+import de.bonndan.nivio.input.dto.LandscapeDescription;
+import de.bonndan.nivio.model.Group;
+import de.bonndan.nivio.model.Item;
+import de.bonndan.nivio.model.LandscapeImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+class GroupQueryResolverTest {
+
+    private GroupQueryResolver groupResolver;
+
+    @Mock
+    private ProcessLog processLog;
+
+    @BeforeEach
+    public void setup() {
+        groupResolver = new GroupQueryResolver(processLog);
+    }
+
+    @Test
+    void process_doesNotContainDuplicatedItemsInGroups() {
+        LandscapeImpl landscape = new LandscapeImpl();
+        landscape.setProcessLog(processLog);
+        Item item = new Item();
+        item.setIdentifier("itemIdentifier");
+        item.setGroup("groupIdentifier");
+        landscape.setItems(Set.of(item));
+        landscape.addGroup(new Group("groupIdentifier"));
+
+        LandscapeDescription input = new LandscapeDescription();
+        GroupDescription groupDescription = new GroupDescription();
+        groupDescription.setIdentifier("groupIdentifier");
+        groupDescription.setContains(List.of("itemidentifier"));
```

Comment:
The query condition has to be the lowercased identifier. Is this intentional? See the other test below.

Reply:
Good point. We must make sure that querying is not case-sensitive.